### DISCUSSION TITLE
Add DigitalContent to OrderLine

### DISCUSSION THREAD

**Discussion by boreBoar** [saleor/saleor#7763] [Q&A] [answered]
Hi!
I'm using Saleor for Digital products.
I generate a dynamic file for each individual customer on a variant they choose.
I don't have automatic fulfillments on for digital products, because i need to run a bunch of system checks before i generate the file. The file is unique to each customer.
However, I can't seem to find a way to add the digital content file once created to the OrderLine.
I can see that the database model in DigitalContentUrl has a foreign key to the OrderLine, but how do i attach the digital content url to the order line through GraphQL?

**Comment by karolkielecki** [comment]
@boreBoar Hi there! I'll check it out with the team and get back to you tomorrow.

**Reply by boreBoar** [reply]
Sure!

**Reply by karolkielecki** [reply]
My colleague, who was working on that, is on holiday. If I have any news, I'll let you know.

**Comment by korycins** [comment, marked as answer]
Hey @boreBoar 
_I can see that the database model in DigitalContentUrl has a foreign key to the OrderLine, but how do i attach the digital content url to the order line through GraphQL?_
I don't think that there is any mutation that will allow you to attach URL to OrderLine. The basic assumption of this feature was that staff user uploads DigitalContent and Saleor generates dedicated URLs for a customer. 
I believe the easiest way would be adding new mutation which will attach digitalcontentUrl to the orderLine, plus setting that will determine if Saleor should generate new URLs.
